### PR TITLE
Issue #424 Add REST api to oauth client id/secret resources for the setup page

### PR DESCRIPTION
**Pull request by chirino**
Also:
- Adds tags field to properties
- Adds configuredProperties to Connectors
- Increments the schema version

### CODE REVIEW

File: rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        List<Connector> items = dataMgr.fetchAll(Connector.class).getItems();
+        items.forEach(connector -> {
+            if (isOauthConnector(connector)) {
+                OAuthApp app = new OAuthApp();
+                app.id = connector.getId().get();
+                app.name = connector.getName();
+                app.icon = connector.getIcon();
+                app.clientId = getPropertyTaggedAs(connector, "oauth-client-id");
+                app.clientSecret = getPropertyTaggedAs(connector, "oauth-client-secret");
```
Comment (zregvart) [inline rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java]: Would this expose secret to the UI? (see  #373)

Comment (chirino) [inline rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java]: yeah. this would need to be addressed as part of #373 too.

File: rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    private final DataManager dataMgr;
+
+    public OAuthAppHandler(DataManager dataMgr) {
+        this.dataMgr = dataMgr;
+    }
+
+    // Since this a a view model DTO, and not a domain model lets define it here instead
+    // of placing it into the model module.
+    public static final class OAuthApp {
```
Comment (jimmidyson) [inline rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java]: Recommend constructor args and getters here.

File: rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    public OAuthAppHandler(DataManager dataMgr) {
+        this.dataMgr = dataMgr;
+    }
+
+    // Since this a a view model DTO, and not a domain model lets define it here instead
+    // of placing it into the model module.
+    @SuppressFBWarnings(
+        value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD",
+        justification = "All fields are encode by jackson and sent to the UI")
+    public static final class OAuthApp {
```
Comment (zregvart) [inline rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java]: Would it make sense to use Immutables nonetheless and have a `from(Connector)` method?

File: rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    private boolean isOauthConnector(Connector connector) {
+        TreeSet EMPTY = new TreeSet();
+        return connector.getProperties().values().stream().anyMatch(x -> {
+                return x.getTags().orElse(EMPTY).contains("oauth-client-id");
+            }
+        );
+    }
+
+    private String getPropertyTaggedAs(Connector connector, String name) {
```
Comment (zregvart) [inline rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java]: This might be useful on the Connector interface also, I think it can be used in Credentials to further simplify the applicators (can be worked on in another PR)
